src/app/collection/settings/collection-settings.component.ts

```diff
@@ -143,34 +141,33 @@ export class CollectionSettingsComponent implements OnInit, OnDestroy {
   }
 
   private subscribeToStore() {
-    this.userCount$ = this.store$.select(selectAllUsers).pipe(map(users => (users ? users.length : 0)));
+    this.userCount$ = this.store$.pipe(
+      select(selectAllUsers),
+      map(users => (users ? users.length : 0))
+    );
 
     const sub1 = this.store$
-      .select(selectWorkspace)
-      .pipe(filter(workspace => !isNullOrUndefined(workspace)))
+      .pipe(
+        select(selectWorkspace),
+        filter(workspace => !!workspace)
+      )
       .subscribe(workspace => (this.workspace = workspace));
     this.subscriptions.add(sub1);
 
     const sub2 = this.store$
-      .select(selectCollectionByWorkspace)
-      .pipe(filter(collection => !isNullOrUndefined(collection)))
-      .subscribe(collection => {
-        this.collection = collection;
-      });
+      .pipe(
+        select(selectCollectionByWorkspace),
+        filter(collection => !!collection)
+      )
+      .subscribe(collection => (this.collection = collection));
     this.subscriptions.add(sub2);
 
-    this.subscriptions.add(
-      this.store$
-        .select(selectPreviousUrl)
-        .pipe(take(1))
-        .subscribe(url => (this.previousUrl = url))
-    );
-
-    this.store$.dispatch(new CollectionsAction.GetNames());
-    this.collectionNames$ = this.store$.pipe(
-      select(selectCollectionNames),
-      withLatestFrom(this.store$.pipe(select(selectCollectionByWorkspace))),
-      map(([names, collection]) => (names && names.filter(name => name !== collection.name)) || [])
-    );
+    const sub3 = this.store$
+      .pipe(
+        select(selectPreviousUrl),
+        take(1)
+      )
+      .subscribe(url => (this.previousUrl = url));
+    this.subscriptions.add(sub3);
```

Comment:
I think you don't really need to unsubscribe observables completed by `take` operator.